pkg/appgw/frontend_listeners.go

```diff
@@ -70,39 +71,24 @@ func (c *appGwConfigBuilder) newListener(listener listenerIdentifier, protocol n
 		Name: to.StringPtr(generateListenerName(listener)),
 		ApplicationGatewayHTTPListenerPropertiesFormat: &n.ApplicationGatewayHTTPListenerPropertiesFormat{
 			// TODO: expose this to external configuration
-			FrontendIPConfiguration: resourceRef(*c.getPublicIPID()),
+			FrontendIPConfiguration: resourceRef(*c.getIPConfigurationID(envVariables)),
 			FrontendPort:            resourceRef(frontendPortID),
 			Protocol:                protocol,
 			HostName:                &listener.HostName,
 		},
 	}
 }
 
-func (c *appGwConfigBuilder) getPublicIPID() *string {
-	var publicIPID *string
-	var jsonConfigs []string
+func (c *appGwConfigBuilder) getIPConfigurationID(envVariables environment.EnvVariables) *string {
+	usePrivateIP, _ := strconv.ParseBool(envVariables.UsePrivateIP)
```

Comment:
I wonder what happens if UsePrivateIP is not provided.
It looks like it would be `""` and most likely default to `False` right?

Reply:
yes, it treats it as `false`